### ISSUE TITLE
[Rule RSRCH]: current rule for tabbable elements not being allowed `none` role

### ISSUE THREAD

**Issue by patrickhlauke** [IBMa/equal-access#1305]
### Rule description

Based on this question in the WCAG repo https://github.com/w3c/wcag/issues/3029 it seems that tabbable elements without a role are being flagged as failures under 4.1.2 (presumably based on the `element_tabbable_role_valid.ts` rule?)

This seems a bit of an overreach/overinterpretation. Related, note that the working groups deemed it valid that arbitrary elements can receive focus (and hopefully this note will be added soon https://github.com/w3c/wcag/pull/1643). 

I'd suggest reviewing the rule, and perhaps softening this from a hard failure of 4.1.2 to a best practice suggestion, or at the very least a manual check?

### Relevant requirement(s)

4.1.2 Name, Role, Value

**Comment by philljenkins** [comment]
I made a similar comment on WCAG repo https://github.com/w3c/wcag/issues/3029  

>If the element doesn't have a valid role, then how is the blind screen reader user going to know how to interact with it?
Name and Role are both required. Depending on the type, then value or state is also required. This Name, Role, and Value was the whole foundation of ARIA and how screen readers work.

>Can you have a button without a name or label? No. It's either a violation of 1.3.1. or 4.1.2
>Can you have a labeled button you can't operate? No, it's a violation of 2.1.1
>Can you have a label/name without a role? No. It's either a violation of 1.3.1. or 4.1.2

>And I flagged static elements that received keyboard focus all the time as a violation because the blind screen reader user has no idea why the keyboard focus stopped there if it's not operable.

**Comment by patrickhlauke** [comment]
I disagree with your take I'm afraid, @philljenkins ... to me, this seems to be an overinterpretation/overreach.

**Comment by philljenkins** [comment]
@patrickhlauke  
I need to walk back a few comments. They are based on the confusion between **_focusable_** and **_tabbable_** - they are NOT the same thing. 

Things that should receive focus that should not be tabbable include local anchors.  

Browsers do not natively add local anchors into the tab order, so they are focusable but not tabbable.

Our rule (and other ACT rules) should be checking that things that are "**_tabbable_**" (in the tab order) should have valid ARIA-defined widget roles.  So the rule should not be failing static content that can receive focus but is not tabbable.  But if some static content is added incorrectly to the tab order it should be flagged.  I think all the examples previously discussed can and should receive focus, but not be in the tab order.  Another example is the tab panel - it receives focus, but its the tabs that would be tabbable. otherwise the panel would be in the tab order and the tabs would be arrowable.

We don't want native tab key users (including voice command & control users, switch users, etc.) stopping on non-interactable content because developers incorrectly think they need to add something to the tab order to make it focusable.

@tombrunet may be able to better articulate the concept and/or disconnect.

**Comment by philljenkins** [comment]
Lets move the discussion to WCAG repo https://github.com/w3c/wcag/issues/3029
and when its resolved there, bring back the resolution here.

**Comment by aliunwala** [comment]
Triage:
IBM rule flags this as a IBM Violation but a WCAG Recomendation, as such we are closing this for now.
